### PR TITLE
Package Common Npm Bearer Auth Mask

### PR DESCRIPTION
**Pull request by tintse-thxsky-MSFT**

### CODE REVIEW

File: common-npm-packages/packaging-common/npm/npmregistry.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export class NpmRegistry implements INpmRegistry {
         this.authOnly = authOnly || false;
     }
 
+    /** Return NpmRegistry with masked auth from Service Endpoint. */
```
Comment (embetten) [inline common-npm-packages/packaging-common/npm/npmregistry.ts:28]: nit: why multiline comment when its a single line?

File: common-npm-packages/packaging-common/npm/npmregistry.ts [hunk 2]
```diff
@@ -59,31 +60,31 @@ export class NpmRegistry implements INpmRegistry {
                 username = endpointAuth.parameters['username'];
                 password = endpointAuth.parameters['password'];
                 email = username; // npm needs an email to be set in order to publish, this is ignored on npmjs
-                password64 = (new Buffer(password).toString('base64'));
```
Comment (embetten) [inline common-npm-packages/packaging-common/npm/npmregistry.ts:62]: why move the secret call from here to below? is this code not working?

File: common-npm-packages/packaging-common/npm/npmregistry.ts [hunk 3]
```diff
@@ -59,31 +60,31 @@ export class NpmRegistry implements INpmRegistry {
...
                 const apitoken = endpointAuth.parameters['apitoken'];
+                tl.setSecret(apitoken);
                 if (!isVstsTokenAuth){
                     // Use Bearer auth as it was intended.
                     auth = nerfed + ':_authToken=' + apitoken + lineEnd;
                 } else {
                     // Azure DevOps does not support PATs+Bearer only JWTs+Bearer
                     email = 'VssEmail';
                     username = 'VssToken';
-                    password64 = (new Buffer(apitoken).toString('base64'));
```
Comment (embetten) [inline common-npm-packages/packaging-common/npm/npmregistry.ts:78]: should this password field be entirely removed ?

File: common-npm-packages/packaging-common/npm/npmregistry.ts (hunk 1 above)
Comment (tintse-thxsky-MSFT) [reply common-npm-packages/packaging-common/npm/npmregistry.ts:28]: I think this is the syntax for method docs.

File: common-npm-packages/packaging-common/npm/npmregistry.ts (hunk 2 above)
Comment (tintse-thxsky-MSFT) [reply common-npm-packages/packaging-common/npm/npmregistry.ts:62]: just refactoring the secret code outside of switch block. cleaner code.

File: common-npm-packages/packaging-common/npm/npmregistry.ts (hunk 3 above)
Comment (tintse-thxsky-MSFT) [reply common-npm-packages/packaging-common/npm/npmregistry.ts:78]: oops I accidently deleted it, added back,

File: common-npm-packages/packaging-common/npm/npmregistry.ts [hunk 4]
```diff
@@ -59,31 +60,32 @@ export class NpmRegistry implements INpmRegistry {
...
-                    tl.setSecret(password64);
+                    password64 = Buffer.from(apitoken).toString('base64');
 
                     auth = nerfed + ':username=' + username + lineEnd;
                     auth += nerfed + ':_password=' + password64 + lineEnd;
                     auth += nerfed + ':email=' + email + lineEnd;
                 }
                 break;
         }
+        tl.setSecret(password);
```
Comment (AlexVTor) [inline common-npm-packages/packaging-common/npm/npmregistry.ts:87]: From the code above, only one of these lines will actually be used in any scenario. Can we use the .setSecret() method on an object that hasn't been set?

Comment (tintse-thxsky-MSFT) [reply common-npm-packages/packaging-common/npm/npmregistry.ts:87]: yea the function setSecret does do a null/empty string check before doing anything.

/**
 * Registers a value with the logger, so the value will be masked from the logs.  Multi-line secrets are not allowed.
 *
 * @param val value to register
 */
function setSecret(val) {
    if (val) {
        if (val.match(/\r|\n/) && ("" + process.env['SYSTEM_UNSAFEALLOWMULTILINESECRET']).toUpperCase() !== 'TRUE') {
            throw new Error(exports.loc('LIB_MultilineSecret'));
        }
        exports.command('task.setsecret', {}, val);
    }
}

**Review by AlexVTor — APPROVED** [review]

**Pull request merged**